qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -144,70 +199,100 @@ class FermionicOp(SecondQuantizedOp):
 
     def __init__(
         self,
-        data: Union[str, Tuple[str, complex], List[Tuple[str, complex]]],
+        data: Union[
+            str,
+            Tuple[str, complex],
+            List[Tuple[str, complex]],
+            List[Tuple[str, float]],
+            List[Tuple[_FermionLabel, complex]],
+        ],
         register_length: Optional[int] = None,
-        sparse_label: bool = False,
+        display_format: str = "dense",
```

Comment:
Can we also add some sort of warning that the default of this is going to change to `sparse` in Qiskit Nature 0.3.0?
Either a `DeprecationWarning` or `UserWarning` should work I think.